save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/controller/HeartbeatController.kt

```diff
@@ -58,21 +57,20 @@ class HeartbeatController(private val agentService: AgentService,
      */
     @PostMapping("/heartbeat")
     fun acceptHeartbeat(@RequestBody heartbeat: Heartbeat): Mono<String> {
-        logger.info("Got heartbeat state: ${heartbeat.state.name} from ${heartbeat.agentId}")
-        areAgentsHaveStarted.compareAndSet(false, true)
-        // store new state into DB
-        return agentService.updateAgentStatusesWithDto(
-            AgentStatusDto(LocalDateTime.now(), heartbeat.state, heartbeat.agentId)
-        )
-            .onErrorResume(WebClientResponseException::class.java) {
-                logger.warn("Couldn't update agent statuses for agent ${heartbeat.agentId}, will skip this heartbeat: ${it.message}")
-                Mono.empty()
+        val executionId = heartbeat.executionProgress.executionId
+        logger.info("Got heartbeat state: ${heartbeat.state.name} from ${heartbeat.agentId} under execution id=$executionId")
+        return {
+            dockerService.markAgentForExecutionAsStarted(executionId)
+            heartBeatInspector.updateAgentHeartbeatTimeStamps(heartbeat)
+        }
+            .toMono()
```

Comment:
why not Mono.just({}) here to make lazy evaluations

Reply:
It's an extension which wraps the lambda into `Mono.fromSupplier`, so it's essentially the same